### DISCUSSION TITLE
Beginner EBSD Reconstruction

### DISCUSSION THREAD

**Discussion by Aventek** [mtex-toolbox/mtex#1420] [Ask Anything]
Hi, I'm relatively new to MTEX and EBSD reconstruction, so apologies if this is a simple question.
I have been trying to do a simple reconstruction of a single slice from a Nickel superalloy dataset (SmallIN100, downloaded from Dream3D Nightly http://dream3d.bluequartz.net/Data/SmallIN100.zip) in MTEX while utilizing Dream3D's example to check that I am getting the correct IPF colors.
The main issue I am running into is that I cannot nail down the filtering parameters required (e.g. smoothing, mori, size) to produce a similar looking IPF map. Below is the best that I am able to get using my attached script, compared to the second image generated following Dream3D's tutorial. I have checked the misorientation between regions that have been differentiated in MTEX but not in Dream3D and they are substantial (>5% limit) so it seems they should be differentiated though the example shows otherwise.

![image](https://user-images.githubusercontent.com/8663880/166338450-69c1bb93-d932-4f29-b49e-b8c1025904f1.png)
![SMALLIN100-IPF-Z_0](https://user-images.githubusercontent.com/8663880/166338661-9bc9250f-6f23-4f24-a164-e66bb1a26cdf.jpg)
`%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3', [3.5 3.5 3.5], 'mineral', 'Nickel', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = 'E:\EBSD\datasets\SmallIN100';

% which files to be imported
fname = [pname '\Slice_1.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame');

mori_lim = 5;
smooth_lim = 15;
size_lim = 5;

[grains,ebsd.grainId, ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',mori_lim*degree)
ori1 = grains(4880).meanOrientation;
ori2 = grains(4735).meanOrientation;
angle(ori1,ori2) ./ degree
ebsd(grains(grains.grainSize<size_lim)) = [];
[grains, ebsd.grainId] = calcGrains(ebsd,'angle',mori_lim*degree)
grains = smooth(grains,smooth_lim);

fig0 = figure('Name', 'Unsmoothened')
plot(ebsd,ebsd.orientations)
hold on
plot(grains.boundary,'linewidth',1.25)
hold off

fig1 = figure('Name','Filled')
ebsd_filled = fill(ebsd,grains);
plot(ebsd_filled('indexed'),ebsd_filled('indexed').orientations);
hold on
plot(grains.boundary,'linewidth',1.25)
hold off
%grains = smooth(grains,smooth_lim,'fill',grains)
%grains = grains(grains.grainSize > size_lim)
%ebsd_map = ebsd(grains)
%ebsd_filled = fill(ebsd_map('indexed'));

fig2 = figure('Name','Grains')
ebsd_filled = fill(ebsd,grains);
plot(grains('indexed'),grains('indexed').meanOrientation);
hold on
plot(grains.boundary,'linewidth',1.25)
hold off

%fig2 = figure('Name','Filtered');
%plot(ebsd_filled, ebsd_filled.orientations)
%hold on
%plot(grains.boundary, 'lineWidth', 1.25)
%hold off
%fig3 = figure('Name','UnFiltered');
%plot(grains,grains.meanOrientation)
`

**Comment by filippeof** [comment]
Hi @Aventek,
Isn't the dream3d data you show an EBSD map rather than a reconstruction of grains?
Dream3d uses the EDAX IPF colorcoding by default. To compare the EBSD data, you could try in mtex:

```matlab
ipfKey = ipfTSLKey(ebsd('Nickel').CS); 
color = ipfKey.orientation2color(ebsd('Nickel').orientations); 
plot(ebsd('Nickel'),color)
```
you might also need to rotate the ipf colorcoding(see: https://mtex-toolbox.github.io/EBSDIPFMap.html#11)

best,
Filippe
